plugins/dell-k2/fu-dell-k2-ec.c

```diff
@@ -44,12 +47,14 @@ typedef struct __attribute__((packed)) { /* nocheck:blocked */
 	guint32 dock_status;
 	guint16 dock_state;
 	guint16 dock_config;
-	guint8 dock_mac_addr[48];
+	guint8 dock_mac_addr[6];
 	guint32 dock_capabilities;
 	guint32 dock_policy;
 	guint32 dock_temperature;
 	guint32 dock_fan_speed;
-	guint8 unused[35];
+	guint16 upf_power;
+	guint8 eppid;
+	guint8 unused[74];
```

Comment:
can all this be migrated to rustgen please? I gave a lot of the existing code a "free pass" with `nocheck:blocked` but I think we should try to enforce rustgen for new code.

Reply:
FWIW I think once we have emulation data in place even for "old" stuff we should migrate the existing code too everywhere we can.

Reply:
maybe fix the data in this change first, and another commit to move to rustgen.

Reply:
Sure do both in the same series.

Reply:
didn't get a full picture on packed elemeents, will need more time on this move.

Reply:
I guess rustgen conversion could be a future improvement as it is way more complicated than my thought, perhaps let the bus go to address some bugs first?

Reply:
The thing is, I think the port to rustgen will actually fix some bugs. e.g. is `module_type` a `uint8_t` or a `uint16_t`? different parts of the codebase are assuming different things, and using an actual type means the compiler does all the hard work for you. I'm happy to explain in more detail what needs to be done if that would be helpful.

Reply:
Added the change to use rust.